contractcourt/channel_arbitrator.go

```diff
@@ -1313,13 +1313,24 @@ func (c *ChannelArbitrator) sweepAnchors(anchors *lnwallet.AnchorResolutions,
 		}
 
 		// If we cannot find a deadline, it means there's no HTLCs at
-		// stake, which means we can relax our anchor sweeping as we
-		// don't have any time sensitive outputs to sweep.
+		// stake, which means we can relax our anchor sweeping
+		// conditions as we don't have any time sensitive outputs to
+		// sweep. However we need to register the anchor output with the
+		// sweeper so we are later able to bump the close fee.
 		if deadline.IsNone() {
 			log.Infof("ChannelArbitrator(%v): no HTLCs at stake, "+
-				"skipped anchor CPFP", c.cfg.ChanPoint)
+				"sweeping anchor with default deadline",
+				c.cfg.ChanPoint)
 
-			return nil
+			// Set the value left for the anchor-cpfp transaction
+			// to the anchor output size in case no hltcs are at
+			// stake. Anchor inputs always have an exclusive group
+			// so they are never grouped with other inputs therefore
+			// even if we would keep this value 0, this anchor will
+			// only be swept if the input params are updated (by
+			// the user) to increase the budget manually. However
+			// using the anchor output is technically more correct.
+			value = 330
```

Comment:
I'm finding it difficult to justify this reassignment to myself. However, I am unsure what the actual problem is. As far as I can tell, the original assignment of `value` is from `findCommitmentDeadlineAndValue` which really totals up the amount of reclaimable value in the htlcs here. In the case of an anchor sweep with no htlc value (and therefore no deadline) it seems like we are just declaring the value to be 300. My question is why isn't it the case that the value returned by `findCommitmentDeadlineAndValue` not already include this 330 sat value? Alternatively, let's say that the `value` field is always only the htlc value, doesn't that mean we should always increment that value by 330 instead of simply reassigning?

Again, not sure what is going on here exactly but this is suspicious to me.

Reply:
Really good remark thank you, I think it should be added to the budget completely without taking any ratios into account. Basically adding it here:

```
	// Calculate the budget based on the value under protection,
		// which is the sum of all HTLCs on this commitment subtracted
		// by their budgets.
		budget := calculateBudget(
			value, c.cfg.Budget.AnchorCPFPRatio,
			c.cfg.Budget.AnchorCPFP,
		)

              // The anchor output can be completely used for this budget without taking any Ratios into account.
              budget += 330

		

		log.Infof("ChannelArbitrator(%v): offering anchor from %s "+
			"commitment %v to sweeper with deadline=%v, budget=%v",
			c.cfg.ChanPoint, anchorPath, anchor.CommitAnchor,
			deadlineDesc, budget)

		// Sweep anchor output with a confirmation target fee
		// preference. Because this is a cpfp-operation, the anchor
		// will only be attempted to sweep when the current fee
		// estimate for the confirmation target exceeds the commit fee
		// rate.
		_, err = c.cfg.Sweeper.SweepInput(
			&anchorInput,
			sweep.Params{
				ExclusiveGroup: &exclusiveGroup,
				Budget:         budget,
				DeadlineHeight: deadlineHeight,
			},
		)


```

Reply:
Yeah I think this is better. Even better would be to make budget assigned once and just add the AnchorOutputValue to the original expression that assigns budget.